### DISCUSSION TITLE
Error Message while fetching despite useSWR and API takes longer then Vercel allows

### DISCUSSION THREAD

**Discussion by t0bbl** [neuefische/questions#598] [Web] [answered]
## 🎉 Introduction:

An error message is displayed even though it should just say "...is loading" 
## 🧮 Pull Request or link to code:

[PR](https://github.com/t0bbl/capstone-project/pull/17)

![Screenshot 2023-05-10 at 13 22 02](https://github.com/neuefische/questions/assets/117731660/9f6aff27-d22f-4a1a-b2f9-6d4148e6dd49)

## ♻️ What are the steps to reproduce this issue?

1. put three Keywords in the Form to send a POST
2. sometimes see "is loading" for a short amount of time
3. get error message
4. reload when fetch is done
5. see the page

## 🤔 What happens?

the form.js sends data to the dall e api and pushes the user to the choosefour page. here 
  if (isLoading || !shirts) {
    return <div>loading...</div>;
  }
should do its magic but the user gets an error message instead. the same happens later on when you try to get Variations.

## 👉 What did you expect to happen?

i expected the useSWR to display the isLoading message while the page waits on a response from the api.

## 🤖 Any logs, error output, etc?

see the screenshot

## ✍️ Any other comments?

The page does NOT work on Vercel since dall e needs more then 5 seconds to answer.
Any suggestions except paying for vercel Pro?

Thanks for having a look.

**Comment by NikRoe** [comment, marked as answer]
We solved the issue regarding the error message directly via Zoom.

To solve the issue, we awaited the `trigger()` when Robin fetched with the `useSWRMutation` hook when requesting variations of an image.

When requesting four images, we solved the issue by changing the order to the post requests and awaiting both of them.
